doc/en/requirements.txt

```
@@ -2,6 +2,6 @@ pallets-sphinx-themes
 pluggy>=1.0
 pygments-pytest>=2.2.0
 sphinx-removed-in>=0.2.0
-sphinx>=3.1,<4
+sphinx>=3.1,<5
```

Maybe bump the minimum as well? 4.2.0 is the first to support Python 3.10:

```suggestion
sphinx>=4.2.0,<5
```

And this can also be removed (need to rebase):

https://github.com/pytest-dev/pytest/blob/b4ab2f094234a0e5a1b14cf8e6d7cd9f48ecbadb/doc/en/requirements.txt#L9-L10

And should we bump Python 3.10 for `docs` and `doctesting` on the CI?

https://github.com/pytest-dev/pytest/blob/b4ab2f094234a0e5a1b14cf8e6d7cd9f48ecbadb/.github/workflows/test.yml#L145-L153